### DISCUSSION TITLE
[Enhancement] Heating the hotend after or while finishing the adaptive mesh

### DISCUSSION THREAD

**Discussion by kilinccagatay** [jschuh/klipper-macros#196] [Ideas]
Is it possible to add an option to heat the extruder after mesh leveling? This is particularly important when using adaptive mesh levelling with an inductive probe since it's positioned really close to the extruder's nozzle. However, this proximity often results in small purges on the bed, which can be described as filament dripping. I believe this issue could be resolved by heating the extruder to its optimal extrusion temperature before starting the printing process.

**Comment by spacepope** [comment]
Hey, i would love to have this too.
Currently i set `variable_start_extruder_probing_temp` to 30 or 40 °C, but one annoying side effect is that i have to wait till cooldown after a filament change 🙄 

So it would be great, if this awesome macro collection becomes even more awesome with such an option 😉

**Comment by kilinccagatay** [comment]
After a year of scraping the build plate before every print, I had a small realization that retracting and extruding some filament before and after probing might help solve this issue. While not a perfect solution, it has drastically reduced filament drips, now down to just one or two on the entire plate. This approach involves two steps. I retract 20 mm of filament and then extrude only 10 mm back, which works well with my Prusa MK3S replica extruder. If you have a longer or shorter nozzle, you may need to adjust these numbers accordingly.

**Steps:**

1. Modifying the Default Machine Start G-code

    I added the commands G1 E-20 F2100 and G1 E10 F2100 before and after the _PRINT_START_PHASE_PROBING phase in the Machine Start G-code. Be sure to check your own Machine Start G-code and copy only the necessary lines from my code below.


```
M190 S0 ; Remove this if autoemit_temperature_commands is off in Prusa Slicer 2.6 and later
M109 S0 ; Remove this if autoemit_temperature_commands is off in Prusa Slicer 2.6 and later
_PRINT_START_PHASE_INIT EXTRUDER={first_layer_temperature[initial_tool]} BED=[first_layer_bed_temperature] MESH_MIN={first_layer_print_min[0]},{first_layer_print_min[1]} MESH_MAX={first_layer_print_max[0]},{first_layer_print_max[1]} LAYERS={total_layer_count} NOZZLE_SIZE={nozzle_diameter[0]}
; Insert custom gcode here.
_PRINT_START_PHASE_PREHEAT
; Insert custom gcode here.
G1 E-20 F2100
_PRINT_START_PHASE_PROBING
G1 E10 F2100
; Insert custom gcode here.
_PRINT_START_PHASE_EXTRUDER
; Insert custom gcode here.
_PRINT_START_PHASE_PURGE
; This is the place to put slicer purge lines if you haven't set a non-zero
; variable_start_purge_length to have START_PRINT automatically calculate and 
; perform the purge (e.g. if using a Mosaic Palette, which requires the slicer
; to generate the purge).
```


2. Adjusting the printer.cfg file

     Add the following line to the printer.cfg file to enable the extruder to heat up, as extruding and retracting require a minimum    temperature:

`variable_start_extruder_preheat_scale: 0.9`



I hope this setup helps others as much as it’s helped me!
Cheers,
Cagatay

**Reply by kilinccagatay** [reply]
@spacepope You might want to try this
